apps/pushy/src/pushy_http_common.erl

```diff
@@ -38,28 +43,93 @@ fetch_authenticated(Path, Headers) ->
         {ok, "404", _ResponseHeaders, _ResponseBody} ->
             not_found;
         {ok, Code, ResponseHeaders, ResponseBody} ->
-            ok = check_http_response(Code, ResponseHeaders, ResponseBody),
+            case check_http_response(Code, ResponseHeaders, ResponseBody) of
+                %% TODO: Temporary shim until refactoring is done
+                ok ->
+                    ok;
+                {error, Reason} ->
+                    throw({error, Reason})
+            end,
             ResponseBody;
         {error, Reason} ->
             throw({error, Reason})
     end.
 
+fetch_unauthenticated(Path) ->
+    Headers = unauthenticated_headers(),
+    Url = unauthenticated_url(Path),
+    parse_chef_api_response(ibrowse:send_req(Url, Headers, get)).
+
+-spec parse_chef_api_response(tuple()) -> chef_api_response() | {error, any()}.
+parse_chef_api_response({ok, Code, Headers, ResponseBody}) ->
+    EJBody = jiffy:decode(ResponseBody),
+    ResponseVersion = response_version_from_headers(Headers),
+    #chef_api_response{ response_api_version = ResponseVersion,
+                        response_code = Code,
+                        response_headers = Headers,
+                        response_body = EJBody};
+parse_chef_api_response({error, Reason}) ->
+    {error, Reason}.
+
+response_version_from_headers([]) ->
+    %% If we get here, we didn't get a version header back,
+    %% assume response is v0.
+    0;
+response_version_from_headers([{"X-Ops-Server-API-Version", VersionJSON}|_T]) ->
+    EJ = jiffy:decode(VersionJSON),
+    %% There are more fields here, but for now we only need this as we won't be retrying
+    Version = ej:get({"response_version"}, EJ),
+    binary_to_integer(Version);
+response_version_from_headers([_H|T]) ->
+    response_version_from_headers(T).
```

Comment:
`response_version_from_headers/1` could be more concisely implemented with `lists:keyfind/3`:

```erlang
case lists:keyfind("X-Ops-Server-API-Version", 1, Headers) of ->
    {"X-Ops-Server-API-Version", VersionJSON} ->
        % do something to return version
        ok;
    false ->
        0
end,
```